core/src/main/scala/org/http4s/jdkhttpclient/JdkWSClient.scala

```diff
@@ -47,115 +48,141 @@ object JdkWSClient {
       jdkHttpClient: HttpClient
   )(implicit F: Async[F]): WSClient[F] =
     WSClient(respondToPings = false) { req =>
-      Dispatcher.sequential.flatMap { dispatcher =>
-        Resource
-          .make {
-            for {
-              wsBuilder <- F.delay {
-                val builder = jdkHttpClient.newWebSocketBuilder()
-                val (subprotocols, hs) = req.headers.headers.partitionEither {
-                  case Header.Raw(ci"Sec-WebSocket-Protocol", p) => Left(p)
-                  case h => Right(h)
+      Dispatcher
+        .sequential(
+          await = true
```

Comment:
This might take care of the dispatcher already shut down exception in the debug logs, haven't verified this yet. I'll check in our setup, we're using the exact same code.

Reply:
might possibly lead to non termination due to https://github.com/rivero-ag/http4s-jdk-http-client/blob/fix-hard-abort-WS-connection/core/src/main/scala/org/http4s/jdkhttpclient/JdkWSClient.scala#L84 and the non-terminating completion stage? at least we are observing `CLOSE_WAIT` sockets again on the client.